scapy/contrib/isotp.py

```diff
@@ -1611,3 +1613,370 @@ def recv(self, x=0xffff):
+            verbose: displays information during scan
+
+    Scan for ISOTP Sockets in the defined range and returns found sockets
+    in a specified format. The format can be:
+    text: human readable output
+    code: python code for copy&paste
+    sockets: if output format is not specified, ISOTPSockets will be
+             created and returned in a list
+    """
+
+    if verbose:
+        print("Filtering background noise...")
+
+    # Send dummy packet. In most cases, this triggers activity on the bus.
+    dummy_pkt = CAN(identifier=0x123,
+                    data=b'\xaa\xbb\xcc\xdd\xee\xff\xaa\xbb')
+
+    background_pkts = sock.sniff(timeout=noise_listen_time,
+                                 started_callback=lambda:
+                                 sock.send(dummy_pkt))
+
+    noise_ids = list(set(pkt.identifier for pkt in background_pkts))
+
+    if extended_addressing:
+        found_packets = scan_extended(sock, scan_range, noise_ids=noise_ids,
+                                      sniff_time=sniff_time, verbose=verbose)
+    else:
+        found_packets = scan(sock, scan_range, noise_ids=noise_ids,
+                             sniff_time=sniff_time, verbose=verbose)
+
+    filter_periodic_packets(found_packets, verbose)
+
+    if output_format == "text":
+        return generate_text_output(found_packets)
+    if output_format == "code":
+        return generate_code_output(found_packets, can_interface)
+    return generate_isotp_list(found_packets, can_interface)
+
+
+def generate_text_output(found_packets):
+    if not found_packets:
+        return "No packets found."
+
+    text = "\nFound %s ISOTP-FlowControl Packet(s):" % str(len(found_packets))
+    for pack in found_packets:
+        extended_id = pack > 0x7ff
+        if extended_id:
+            send_id = int(pack / 256)
+            send_ext = pack - (send_id * 256)
+            ext_id = hex(orb(found_packets[pack][0].data[0]))
+            text += "\nSend to ID:             %s" \
+                    "\nSend to extended ID:    %s" \
+                    "\nReceived ID:            %s" \
+                    "\nReceived extended ID:   %s" \
+                    "\nMessage:                %s" % \
+                    (hex(send_id), hex(send_ext),
+                     hex(found_packets[pack][0].identifier), ext_id,
+                     repr(found_packets[pack][0]))
+        else:
+            text += "\nSend to ID:             %s" \
+                    "\nReceived ID:            %s" \
+                    "\nMessage:                %s" % \
+                    (hex(pack),
+                     hex(found_packets[pack][0].identifier),
+                     repr(found_packets[pack][0]))
+
+        padding = found_packets[pack][0].length == 8
+        if padding:
+            text += "\nPadding enabled"
+        else:
+            text += "\nNo Padding"
+
+        text += "\n"
+    return text
+
+
+def generate_code_output(found_packets, can_interface):
```

Comment:
Why would you do something like this ?
It's pretty ugly (especially later, when you start to call `eval()` on it).
It sounds really unsafe :/ It's probably even exploitable if you ask me

Reply:
In the past, it was useful for us, to save the results of a scan in this format in a log file. 
At a later point during a pen-test, we could copy and paste the socket description onto a scapy interactive session, to run further tests. 
This output was not intend to be used in an automated way, where a exploit of this output would be likely.

Reply:
Alright !

Reply:
I double checked the `generate_code_output`. I don't think that this function is exploitable after my latest changes. At the moment, I still think that this function is a useful feature for us. If you are ok with that function, I would like to keep it.

Reply:
Could you add a doctring?